smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java

```diff
@@ -165,16 +169,20 @@ private String serialize(Model fullModel, Collection<Shape> shapes) {
                 continue;
             }
             OperationShape operation = shape.asOperationShape().get();
-            operation.getInput().ifPresent(shapeId -> {
-                if (shapes.contains(fullModel.expectShape(shapeId))) {
-                    inlineableShapes.add(shapeId);
+            if (!operation.getInputShape().equals(UnitTypeTrait.UNIT)) {
+                Shape inputShape = fullModel.expectShape(operation.getInputShape());
+                if (shapes.contains(inputShape) && inputShape.hasTrait(InputTrait.ID)
+                        && operation.getInputShape().getName().equals(operation.getId().getName() + "Input")) {
```

Comment:
Looks like this is only going to inline shapes with the default suffix control statements. It'd bring a lot of complexity to try and set these statements, but calling it for confirmation.

Reply:
Yes. There's already plenty of complexity without introducing that too.